internal/smartlink/smartlink_test.go

```diff
@@ -0,0 +1,75 @@
+package smartlink
+
+import (
+	"os"
+	"path/filepath"
+	"runtime"
+	"testing"
+
+	"github.com/ActiveState/cli/internal/fileutils"
+	"github.com/stretchr/testify/require"
+)
+
+func TestLinkContentsWithCircularLink(t *testing.T) {
```

Comment:
Can we also add a test for the issue that brought up this bug? I imagine we have a commit for the project that was surfacing the bug, we should be able to use that.

Reply:
We could, but it's a massive runtime that would really bog down CI. I think the unit test is sufficient.